File: 3429-special-array-i/3429-special-array-i.py
class Solution:
    def checking(self,num):
        if num%2 == 0:
            return True
        else:
            return False

    def isArraySpecial(self, nums: List[int]) -> bool:
        if len(nums)==1:
            return True
        i = 1
        while i<len(nums):
            if self.checking(nums[i-1]) == self.checking(nums[i]):
                return False
            i+=1
        
        return True
        

File: 3429-special-array-i/3429-special-array-i.java
class Solution {
    public boolean isArraySpecial(int[] nums) {
        if(nums.length==1){
            return true;
        }
        int i = 1;
        while(i<nums.length){
            if(checking(nums[i-1]) == checking(nums[i])){
                return false;
            }
            i+=1;
        }
        return true;
    }
    private boolean checking(int num){
        if(num%2 == 0){
            return true;
        }else{
            return false;
        }
    }
}